### PR TITLE
wifi: shell: Add band support for STA

### PR DESCRIPTION
**Pull request by krish2718**
For a Wi-Fi station the connect API supports both band and channel configuration, but for a shell command either channel or band makes sense, so, overload the channel field to support band.

Rejig the band and channel validation to support all modes.

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -478,32 +485,49 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
...
+							band ? "," : "",
+							wifi_band_txt(all_bands[band]));
 				if (offset >= sizeof(bands_str)) {
 					print(context.sh, SHELL_ERROR,
-					      "Failed to parse channel: %s: "
-					      "band string too long\n",
-					      argv[idx]);
+						"Failed to parse channel: %s: "
+						"band string too long\n",
+						argv[idx]);
```
Comment (sachinthegreen) [inline subsys/net/l2/wifi/wifi_shell.c:526]: Are these changes necessary? If so, maybe they should be part of a separate commit?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -512,12 +536,11 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
 
 			if (!found) {
 				print(context.sh, SHELL_ERROR,
-				      "Invalid channel: %ld, checked bands: %s\n",
-				       channel,
-				       bands_str);
+					"Invalid channel: %ld, checked bands: %s\n",
+					channel,
+					bands_str);
```
Comment (sachinthegreen) [inline subsys/net/l2/wifi/wifi_shell.c:541]: Are these changes necessary? If so, maybe they should be part of a separate commit?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 3]
```diff
@@ -512,12 +536,11 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
...
 				return -EINVAL;
 			}
-
```
Comment (sachinthegreen) [inline subsys/net/l2/wifi/wifi_shell.c:520]: Maybe this should also be clubbed as part of the separate commit?

**Review by sachinthegreen — APPROVED** [review]

File: subsys/net/l2/wifi/wifi_shell.c [hunk 4]
```diff
@@ -478,32 +485,49 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
...
-			const uint8_t bands[] = {WIFI_FREQ_BAND_2_4_GHZ,
-					       WIFI_FREQ_BAND_5_GHZ,
-					       WIFI_FREQ_BAND_6_GHZ};
-			uint8_t band;
-			bool found = false;
-			char bands_str[MAX_BANDS_STR_LEN] = {0};
-			size_t offset = 0;
-
-			for (band = 0; band < ARRAY_SIZE(bands); band++) {
+			if (channel < 0) {
```
[the remark this answers is not in the document]
Comment (krish2718) [reply subsys/net/l2/wifi/wifi_shell.c:508]: 0 is a valid channel i.e., ANY

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
